starknet/rust/src/lib.rs

```diff
@@ -1,34 +1,36 @@
-use std::ffi::{c_char, CStr, CString};
 use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use std::ffi::{c_char, CStr, CString};
 
 #[no_mangle]
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub extern "C" fn compileSierraToCasm(sierra_json: *const c_char) -> *mut c_char {
     let sierra_json_str = unsafe { CStr::from_ptr(sierra_json) }.to_str().unwrap();
 
-    let sierra_class =
-        serde_json::from_str(sierra_json_str).map_err(|err| err.to_string());
+    let sierra_class = serde_json::from_str(sierra_json_str).map_err(|err| err.to_string());
     if let Err(e) = sierra_class {
-        return raw_cstr(e)
+        return raw_cstr(e);
     }
 
-    let casm_class = CasmContractClass::from_contract_class(sierra_class.unwrap(), true, usize::MAX)
-        .map_err(|err| err.to_string());
+    let casm_class =
+        CasmContractClass::from_contract_class(sierra_class.unwrap(), true, usize::MAX)
```

Comment:
we should rewrite `if let` above to match stmt. so we won't call `sierra_class.unwrap()` here
hint: you can shadow existing variable with the same name in Rust:
```
let value = 111; // int
let value = value > 10; // bool
```